frontend/ui/uimanager.lua

```diff
@@ -47,8 +48,42 @@ function UIManager:init()
         -- resume.
         self:_initAutoSuspend()
         self.event_handlers["Suspend"] = function(input_event)
-            self:_stopAutoSuspend()
-            Device:onPowerEvent(input_event)
+            if not self._power_off_triggered then
```

Comment:
Oh, yeah, good point. We can reset the Suspend handler in cancel_callback in this case.

Now that I think I have a better idea on how to implement this. Instead of emitting suspend event, we can just emit PowerRelease event. Then in UIManager, we remember the time for the PowerPress event. In PowerRelease's handler, we check against the press time to decide what action to take. Just like @Hzj-jie implements the autosuspend feature. This way, we can easily add more actions in the future, for example, hard reboot if hold for 10s.